Installing-and-Configuring/Upgrading.md

```diff
@@ -19,10 +19,7 @@ since we don't have a fail-safe recovery mechanism in the event of
 a bad flash or buggy firmware.
 
 Additionally, *reflashing the firmware will change the TPM PCRs*.
-This will require generating a new TPM TOTP token and a new drive
-encryption key.  Be sure you have your TPM owner's password and your
-disk encryption recovery key or passphrase available since, by design,
-the disk key is not accessible to the recovery shell.
+This will require generating a new TPM TOTP token.  If you stored the decryption key for your drive in the TPM it will be lost.  Be sure you have your TPM owner's password and your disk encryption recovery key or passphrase available since the disk key is not accessible to the recovery shell.  This is by design.
```

Comment:
Disk Unlock Key will need to be renewed since not valid in TPM nv space. Reseal HOTP/TOTP, sign /boot and setup a new boot default and you will be good to go.
Do it from the menus. not the shell.

Please remove fear langage and replace by proper instructions on what should be done, not what should not be done.

Reply:
WIP  reworded.  will be pushed in a few minutes.

Reply:
Disk Recovery Key is the key that was setuped at OS installation with relative passphrase, this setuped slot 0.

Disk Unlock Key was setuped randomly in Heads when setuping a boot default on boards having a TPM and not having deactivated the option in board config. The Disk Unlock Key passphrase unseals the TPM Disk Unlock Key if NV regions Disk Unlock Key passphrase is valid and measurements are coherent with same modules and measured content in PCRs then when the secret was sealed. 

The user can setup and change the Disk Unlock Key passphrase but setting a new boot default anytime, and providing Disk Recovery Key passphrase, new Disk Unlock Key passphrase and typing GPG User PIN to detach sign new boot config.

That process should be redo everytime boot content is changed. Meaning the user should reboot after updating dom0 under Qubesos or OS under monolitic OSes to attest boot related changes ( kernel initrd xen grub)

Reply:
reworded to use original text with a few clarifications and consistent terminology